packages/deploy/src/stateTransform.ts

```diff
@@ -149,6 +178,16 @@ function mergeTriggers(
               ...(specTrigger!.type === 'cron'
                 ? { cron_expression: specTrigger!.cron_expression }
                 : {}),
+              ...(specTrigger!.type === 'kafka'
+                ? {
+                    kafka_configuration: {
+                      ...specTrigger!.kafka_configuration,
+                      hosts: transformSpecKafkaHost(
+                        specTrigger!.kafka_configuration?.hosts ?? []
```

Comment:
I would default the argument of `transformSpecKafkaHost` inside the function declaration, rather than passing an empty array here:

```
function transformSpecKafkaHost(hosts: string[] = []): StateKafkaHost[] { ... }
```